src/main/shadow/cljs/ui/views/runtimes.cljs

```diff
@@ -0,0 +1,141 @@
+            [:p.text-gray-900.font-medium (when host (name host))] (or desc user-agent "")]
+           [:td.px-6.py-4.text-right.whitespace-nowrap.text-sm.text-gray-500 (age-display since)]]))))
+
+(defn ui-runtime-listing [runtimes]
+  (<< [:table.min-w-full.divide-y.divide-gray-200
+       [:thead
+        [:tr
+         [:th.px-6.py-3.bg-gray-50.text-left.text-xs.font-medium.text-gray-500.uppercase.tracking-wider "build"]
+         [:th.px-6.py-3.bg-gray-50.text-right.text-xs.font-medium.text-gray-500.uppercase.tracking-wider "host"]
+         [:th.px-6.py-3.bg-gray-50.text-right.text-xs.font-medium.text-gray-500.uppercase.tracking-wider "since"]]]
+       [:tbody.bg-white.divide-y.divide-gray-200
+        (sg/render-seq runtimes identity ui-runtime-overview)]]))
+
+(defc ui-runtime-overview-mobile
+  [ident]
+  (bind {:keys [runtime-id runtime-info supported-ops] :as data}
+    (sg/query-ident ident
+      [:runtime-id
+       :runtime-info
+       :supported-ops]))
+  (render
+    (let [{:keys [lang build-id host type since user-agent desc]} runtime-info]
+      (<< [:li
+           [:a.block.px-4.py-4.bg-white.hover:bg-gray-50 {:href (if build-id
+                                                                  (str "/build/" (name build-id))
+                                                                  "#")}
+            [:span.flex.items-center.space-x-4
+             [:span.flex-1.flex.space-x-2.truncate
+              (build-lang lang)
+              [:span.flex.flex-col.text-gray-500.text-sm.truncate
+               [:span.truncate
+                [:div.inline-flex.space-x-2
+                 [:p.text-gray-500.truncate.group-hover:text-gray-900 runtime-id]
+                 [:p.text-gray-500.truncate.group-hover:text-gray-900 (when build-id (name build-id))]]]
+               [:p.text-gray-900.font-medium (when host (name host))] (or desc user-agent "")
+               [:span (age-display since)]]]
+             [:svg.flex-shrink-0.h-5.w-5.text-gray-400 {:xmlns "http://www.w3.org/2000/svg" :viewBox "0 0 20 20" :fill "currentColor" :aria-hidden "true"}
+              [:path {:fill-rule "evenodd" :d "M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z" :clip-rule "evenodd"}]]]]]))))
+
+(defn ui-runtime-listing-mobile
+  [runtimes]
+  (<< [:ul.mt-2.divide-y.divide-gray-200.overflow-hidden.shadow.sm:hidden
+       (sg/render-seq runtimes identity ui-runtime-overview-mobile)]))
+
+(defc ui-page []
+  (bind {::m/keys [runtimes-sorted]}
+    (sg/query-root
+      [::m/runtimes-sorted]))
+
+  (render
+    (<<
+      [:div.shadow.sm:hidden.relative.h-auto.overflow-y-auto
+       (ui-runtime-listing-mobile runtimes-sorted)]
+      [:div.hidden.sm:block
+       [:div.max-w-7xl.mx-auto
+        [:div.flex.flex-col.mt-2
+         [:div.align-middle.min-w-full.overflow-x-auto.shadow.overflow-hidden.xl:rounded-lg
+          (ui-runtime-listing runtimes-sorted)]]]])))
```

Comment:
This type of rendering makes less sense for client-side JS. We can check the viewport size and render accordingly ourselves. Do not need to render both and let CSS decide. I can do this optimization later though.

The "desktop" variant however looks bad now. Columns squished to much.

Remove "rounded-lg" unless you plan on adding a whole bunch of rounded anywhere else :)

Reply:
If I remove `"round-lg"` for `xl:` then the corners are not rounded when the table is not snuggling with the window.